core/js/src/test/scala/com/lynbrookrobotics/potassium/JSClockTest.scala

```diff
@@ -42,4 +42,16 @@ class JSClockTest extends AsyncFunSuite {
 
     future.map(t => assert(t <= 250))
   }
+
+  test("Single execution can be canceled") {
+    val testDonePromise = Promise[Long]()
+    val future = testDonePromise.future
+    var executed = false
+
+    JSClock.singleExecution(Milliseconds(500)) {
+      throw new IllegalStateException("This code should not have run")
+    }()
+
+    future.map { _ => assert(true) }
```

Comment:
This still doesn't look right. When is the `testDonePromise`, which `future` depends on, completed?